### ISSUE TITLE
no such file to load -- lib/cucumber/nagios/version

### ISSUE THREAD

**Issue by hedgehog** [auxesis/cucumber-nagios#41]
This may be a Ruby 1.9/Rubygems thing...

```
$ git clone git@github.com:hedgehog/cucumber-nagios.git
$ pushd cucumber-nagios/
$ rake -T --trace
(in /usr/src/cucumber-nagios)
rake aborted!
no such file to load -- lib/cucumber/nagios/version
<internal:lib/rubygems/custom_require>:29:in `require'
<internal:lib/rubygems/custom_require>:29:in `require'
/usr/src/cucumber-nagios/Rakefile:5:in `<top (required)>'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2383:in `load'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2383:in `raw_load_rakefile'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2017:in `block in load_rakefile'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2068:in `standard_exception_handling'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2016:in `load_rakefile'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2000:in `block in run'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:2068:in `standard_exception_handling'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/lib/rake.rb:1998:in `run'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/gems/rake-0.8.7/bin/rake:31:in `<top (required)>'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/bin/rake:19:in `load'
/home/hedge/.rvm/gems/ruby-1.9.2-p136@bbb-dev/bin/rake:19:in `<main>'
$ ls -la lib/cucumber/nagios/version.rb 
-rw-r--r-- 1 hedge src 65 2011-02-23 21:35 lib/cucumber/nagios/version.rb
```

Best practice is to remove rubygems - end users should require it if that is how they manage their Gems - considered impolite for lib devs to impose their choices... :)

```
$ git diff
diff --git a/Rakefile b/Rakefile
index b2044ad..6484c2d 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,8 +1,7 @@
 #!/usr/bin/env ruby

-require 'rubygems'
 require 'rake'
-require 'lib/cucumber/nagios/version'
+require './lib/cucumber/nagios/version'

 begin
   require 'jeweler'
```

**Comment by josephholsten** [comment]
+1 to removing `require 'rubygems'` and managing my own load path.

As jeweler demonstrates, it's easy to newbie-proof this. Something like:

```
begin
  require 'cucumber/nagios/version'
rescue LoadError
  warn 'Could not load cucumber/nagios/version. Is your load path setup?  To always have rubygems manage your load path, try running: export RUBYOPT=rubygems.'
end
```

Also, if you're going to use a relative path, you need to use `__FILE__`, not assume that `.` is the same directory as the file you're in. So this might be

```
begin
  require 'cucumber/nagios/version'
rescue LoadError
  warn 'Could not load cucumber/nagios/version. Is your load path setup? I'm assuming you want to use rubygems. To always have rubygems manage your load path, try running: export RUBYOPT=rubygems.'
  require 'rubygems'
  lib_dir = File.expand_path(__FILE__, '../lib')
  $LOAD_PATH.unshift(lib_dir) unless $LOAD_PATH.include(lib_dir)
  require 'cucumber/nagios/version'
end
```

**Comment by auxesis** [comment]
Thanks for your patches, all! 

The fix has been released in 0.9.0.
